modules/@apostrophecms/asset/index.js

```diff
@@ -56,65 +142,56 @@ module.exports = {
           // Don't clutter up with previous builds.
           await fs.remove(buildDir);
           await fs.mkdirp(buildDir);
-          await fs.remove(modulesDir);
-          await fs.mkdirp(modulesDir);
 
-          let rebuildAposUi = argv && !argv['check-ui-build'];
+          for (const [ name, options ] of Object.entries(self.options.builds)) {
+            let rebuild = argv && !argv['check-ui-build'];
 
-          const APOS_MERGED_BUNDLE = 'apos-bundle.js';
-          const APOS_ONLY_BUNDLE = 'apos-only-bundle.js';
-          const APOS_ONLY_TS = '.apos-only-timestamp.txt';
-          const PUBLIC_BUNDLE_CSS = 'public-bundle.css';
-          const PUBLIC_BUNDLE_JS = 'public-bundle.js';
-          let checkTimestamp = false;
+            let checkTimestamp = false;
 
-          const bundleExists = await fs.pathExists(bundleDir);
+            if (options.core) {
+              const bundleExists = await fs.pathExists(bundleDir);
 
-          if (!bundleExists) {
-            rebuildAposUi = true;
-            await fs.mkdirp(bundleDir);
-          } else {
-            await fs.remove(`${bundleDir}/${APOS_MERGED_BUNDLE}`);
-            await fs.remove(`${bundleDir}/${PUBLIC_BUNDLE_CSS}`);
-            await fs.remove(`${bundleDir}/${PUBLIC_BUNDLE_JS}`);
-          }
+              if (!bundleExists) {
+                rebuild = true;
+              }
 
-          if (!process.env.CORE_DEV) {
-            checkTimestamp = await fs.pathExists(`${bundleDir}/${APOS_ONLY_TS}`);
-          }
+              if (!process.env.CORE_DEV) {
+                checkTimestamp = await fs.pathExists(`${bundleDir}/${name}-only-timestamp.txt`);
+              }
+
+              if (!rebuild && checkTimestamp) {
+                // If we have a UI build timestamp file compare against the app's
+                // package.json modified time.
+                if (await lockFileIsNewer(name)) {
+                  rebuild = true;
+                }
+              } else {
+                rebuild = true;
+              }
+            }
 
-          if (!rebuildAposUi && checkTimestamp) {
-            // If we have a UI build timestamp file compare against the app's
-            // package.json modified time.
-            if (await lockFileNewerThanUi()) {
-              rebuildAposUi = true;
-              await fs.remove(`${bundleDir}/${APOS_ONLY_BUNDLE}`);
+            if (rebuild) {
+              await fs.mkdirp(bundleDir);
+              await build(name, options);
             }
-          } else {
-            rebuildAposUi = true;
-            await fs.remove(`${bundleDir}/${APOS_ONLY_BUNDLE}`);
           }
 
-          await moduleOverrides();
-          await buildPublicCssBundle();
-          buildPublicJsBundle();
-
-          if (rebuildAposUi) {
-            await buildAposBundle();
+          const scenes = [...new Set(Object.values(self.options.builds).map(options => options.scenes).flat()) ];
```

Comment:
Identify the unique scenes (just "public" and "apos" for now and for the foreseeable future, but we're not hardcoding that).